src/convnets/densenet.jl

```diff
@@ -1,20 +1,32 @@
-"""
-    dense_bottleneck(inplanes, outplanes; expansion=4)
+function dense_bottleneck(inplanes::Integer, growth_rate::Integer, bn_size::Integer,
+                          dropout_prob)
+    return Chain(cat_channels,
+                 conv_norm((1, 1), inplanes, bn_size * growth_rate;
+                            revnorm = true)...,
+                 conv_norm((3, 3), bn_size * growth_rate, growth_rate;
+                            pad = 1, revnorm = true)...,
+                 Dropout(dropout_prob))
+end
 
-Create a Densenet bottleneck layer
-([reference](https://arxiv.org/abs/1608.06993)).
+function dense_block(inplanes::Integer, num_layers::Integer, bn_size::Integer,
+                     growth_rate::Integer, dropout_prob)
+    layers = [dense_bottleneck(inplanes + (i - 1) * growth_rate, growth_rate, bn_size,
+                                   dropout_prob) for i in 1:num_layers]
+    return DenseBlock(layers)
+end
 
-# Arguments
+struct DenseBlock
+    layers::Any
+end
+@functor DenseBlock
 
-  - `inplanes`: number of input feature maps
-  - `outplanes`: number of output feature maps on bottleneck branch
-    (and scaling factor for inner feature maps; see ref)
-"""
-function dense_bottleneck(inplanes::Int, outplanes::Int; expansion::Int = 4)
-    return SkipConnection(Chain(conv_norm((1, 1), inplanes, expansion * outplanes;
-                                          revnorm = true)...,
-                                conv_norm((3, 3), expansion * outplanes, outplanes;
-                                          pad = 1, revnorm = true)...), cat_channels)
+function (m::DenseBlock)(x)
+    input = [x]
+    for layer in m.layers
+        x = layer(input)
+        input = vcat(input, [x])
+    end
+    return cat_channels(input...)
 end
```

Comment:
This is where I think the biggest change is, and where I think the code could use the most input. Is this the best way of doing this sequence of operations?

Reply:
maybe 
```julia
function (m::DenseBlock)(x)
    input = (x,)
    for layer in m.layers
        x = layer(input)
        input = (input..., x)
    end
    return cat_channels(input)
end
```
to bypass the Zygote `vcat` problem?

Reply:
Even better, we can remove `cat_channels` from the layers and do 
```julia
function (m::DenseBlock)(x)
    input = x
    for layer in m.layers
        x = layer(input)
        input = cat_channels(input, x)
    end
    return input
end